android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1784,6 +1789,68 @@ public Profile getCurrentProfile() {
         return Profile.fromWebContents(tab.getWebContents());
     }
 
+    /** Close all tabs whose URL origin matches with a given origin. */
+    public void closeAllTabsByOrigin(@NonNull final String origin) {
+        final Tab activeTab = getActivityTab();
+        final TabModel tabModel = getCurrentTabModel();
+        if (activeTab != null && activeTab.getUrl().getOrigin().getSpec().equals(origin)) {
+            activeTab.setClosing(true);
+            tabModel.closeTab(activeTab);
+        }
+
+        Set<Integer> tabIndexes = getTabIndexesByUrlOrigin(tabModel, origin);
+        for (Integer index : tabIndexes) {
+            Tab tab = tabModel.getTabAt(index);
+            if (tab != null) {
+                tab.setClosing(true);
+                tabModel.closeTab(tab);
+            }
+        }
+    }
+
+    public Tab selectExistingUrlOriginTab(@NonNull final String origin) {
+        Tab tab = getActivityTab();
+        if (tab != null && tab.getUrl().getOrigin().getSpec().equals(origin)) {
+            return tab;
```

Comment:
the same questions about the tab indexes Set

Reply:
Changes done in commit [`91727f8` (#24960)](https://github.com/brave/brave-core/pull/24960/commits/91727f880a0647d3271072b698cfd760dddad24d)